src/bootstrap/test.rs

```diff
@@ -514,6 +514,41 @@ impl Step for RustdocJS {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocUi {
```

Comment:
This struct should be replaceable with a `host_test!` macro invocation I believe.

Reply:
I tried to at first to avoid writing code by hand, but failed... I can't only rely on UI or rustdoc test suite, I need a mix of the two.